capstone/src/main/java/com/google/sps/search/Trie.java

```diff
@@ -1,35 +1,122 @@
 package com.google.sps.search;
 
 import java.io.Serializable;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
+import java.util.TreeSet;
 
+/**
+ * Class representing the Trie data structure. Contains children to indicate child nodes in the data
+ * structure and whether current node is the end of the name.
+ *
+ * @author lucyqu
+ */
 public class Trie implements Serializable {
 
   private static final long serialVersionUID = 1L;
   private Map<String, Trie> children;
-  private boolean isEnd;
+  private boolean isName;
+  private boolean isFullName;
 
+  /**
+   * Constructor that sets the children of trie node and whether current node marks the end of the
+   * name.
+   */
   public Trie() {
-    children = new HashMap<String, Trie>();
-    isEnd = false;
+    this.children = new HashMap<String, Trie>();
+    this.isName = false;
+    this.isFullName = false;
   }
 
+  /**
+   * Inserts name into trie, with full name being the last node appended to the end of the name.
+   *
+   * @param name user first name or last name
+   * @param fullName user full name.
+   */
   public void insert(String name, String fullName) {
     if (name.contentEquals("")) {
-      isEnd = true;
-      // add in the full name here
+      // indicates the end of first name or last name
+      this.isName = true;
+      // insert full name node as child
+      Trie fullNameNode = new Trie();
+      fullNameNode.isFullName = true;
+      this.children.put(fullName, fullNameNode);
```

Comment:
I see what you're doing, but this seems like an odd overloading of existing features.

I would suggest you keep the trie->child relation ship always 1 lowercase character. This is the kind of feature that breaks when a user has a 1 letter first or last name in the DB or something odd.

Instead, I would remove `isFullName` and replace it with `Set<String> fullNames` so you can just check if the set is empty, or get all unique full names.